Opencast 2853 transcript docx available in engage downloads

npmali opened the pull request:
### Your pull request should…

* [ ] have a concise title
* [ ] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [ ] be against the correct branch (features can only go into develop)
* [ ] include migration scripts and documentation, if appropriate
* [ ] pass automated testing
* [ ] have a clean commit history
* [ ] have proper commit messages (title and body) for all commits
* [ ] have appropriate tags applied

npmali committed:
OPENCAST-2853: Initial commit

- Extracted and published the docx file on nibity trancription java code
- retrieved docx file in download lti home page
- retrieved docx file in download plugin

npmali committed:
OPENCAST-2853: Removed extra brace

npmali committed:
OPENCAST-2853: formatted code

npmali committed:
OPENCAST-2853: Moved variable declarations to separate lines

npmali committed:
OPENCAST-2853: cleanup

npmali committed:
OPENCAST-2853: Added a check for both vtt and docx file and cleanup

npmali committed:
OPENCAST-2853: declared WorkflowOperationException

npmali committed:
OPENCAST-2853: Added whitespace before curly brace

npmali committed:
OPENCAST-2853: Fixed captions download formats

npmali committed:
Merge branch 'r/8.x' into OPENCAST-2853-Transcript_docx_available_in_Engage_Downloads

npmali committed:
OPENCAST-2853: Changed captions format on lti and engage player

npmali committed:
OPENCAST-2853: Changed captions format and download URL

npmali committed:
OPENCAST-2853: Cleanup

npmali committed:
Merge branch 'r/8.x' into OPENCAST-2853-Transcript_docx_available_in_Engage_Downloads

npmali committed:
OPENCAST-2853: Cleanup

smarquard commented on modules/transcription-service-workflowoperation/src/main/java/org/opencastproject/transcription/workflowoperation/NibityAttachTranscriptionOperationHandler.java:
```diff
@@ -175,4 +172,35 @@ public void setWorkspace(Workspace service) {
     this.workspace = service;
   }
 
+  public MediaPackage addTranscriptToMediaPackage(InputStream zis, String captionMimeType, String captionIdentifier, String captionFileType, MediaPackage mediaPackage, MediaPackageElementFlavor flavor, String targetTagOption) throws WorkflowOperationException {
```
This is used both to add a transcript and captions so it would be clearer to give it a more generic name, e.g. addElementToMediaPackage or addTranscriptionElementToMediaPackage

smarquard replied:
Also avoid super-long lines.

smarquard commented on modules/transcription-service-workflowoperation/src/main/java/org/opencastproject/transcription/workflowoperation/NibityAttachTranscriptionOperationHandler.java:
```diff
@@ -121,45 +121,42 @@ public WorkflowOperationResult start(final WorkflowInstance workflowInstance, Jo
+
+      if (zippedVtt == null && zippedDocx == null) {
+        logger.debug("No captions found in results zip file {}", transcription.getURI());
+        throw new WorkflowOperationException("No captions found in the zip file");
```
The debug and exception text here is not accurate; neither captions nor transcript found in the zip file is better.

smarquard commented on modules/transcription-service-workflowoperation/src/main/java/org/opencastproject/transcription/workflowoperation/NibityAttachTranscriptionOperationHandler.java:
```diff
@@ -175,4 +172,35 @@ public void setWorkspace(Workspace service) {
+
+      // Set the target flavor if informed
+      if (flavor != null)
+        transcriptElement.setFlavor(flavor);
```
Always use {} for if statements as a style issue.

npmali committed:
Merge branch 'r/8.x' into OPENCAST-2853-Transcript_docx_available_in_Engage_Downloads

npmali committed:
OPENCAST-2853: Changed nibity API endpoint to test endpoint

npmali committed:
OPENCAST-2583: Changed test API endpoint to nibity

Also cleaned the code:
- added {} to if statements
- changed debug exception text
- broken long lines of code
- changed function name from "addTranscriptToMediaPackage" to  "addTranscriptionElementToMediaPackage"
